ion-hash/tests/ion_hash_tests.rs

```diff
@@ -1,71 +1,96 @@
 // Copyright Amazon.com, Inc. or its affiliates.
 
-use ion_hash::{self, IonHashDigest, IonHasher};
+use digest::{consts::U256, generic_array::GenericArray, Digest, Output};
+use ion_hash::{self, IonHasher};
 use ion_rs::result::IonResult;
 use ion_rs::value::loader::{loader, Loader};
 use ion_rs::value::*;
-use std::{
-    fs::read,
-    sync::{Arc, Mutex},
-};
+use std::fs::read;
 
 #[derive(Default, Clone)]
 struct TestDigest {
-    inner: Arc<Mutex<TestDigestInner>>,
+    updates: GenericArray<u8, U256>,
+    position: usize,
 }
```

Comment:
Is this the identity digest?  If so, documenting this would be nice.

Also, couldn't you just use a `Vec` for this?

Reply:
Yes. Will add a comment.

The `Digest` trait uses fixed size arrays. So, we have to play by that if we want to use that trait (which I think we should). While the test digest could use a Vec internally, it would still need to turn it into a generic array as part of the final result. And, that conversion has to match exactly (the vec slice len has to be exactly the len of the associated output size). So you land up with annoying code at some point. I started with Vec (see previous commits) and moved to this because I felt it was simpler.

Reply:
I could use a Vec internally only *and test against it* if we're prepared to do the interior mutability trick I had in a previous version. But.. I actually quite like what I have here. Let me know what you think.